cluster/cluster.go

```diff
@@ -79,20 +79,31 @@ func Start() {
 	log.Info("CLU Start: joined to %d nodes in cluster", n)
 }
 
+type partitionCandidates struct {
+	priority int
+	nodes    []Node
+}
+
 // return the list of nodes to broadcast requests to
-// Only 1 member per partition is returned. This list includes
-// ThisNode if it is capable of handling queries.
+// Only 1 member per partition is returned.
+// The nodes are selected based on priority, using a random selection
+// if there are multi nodes with the highest priority.
 func MembersForQuery() []Node {
 	thisNode := Manager.ThisNode()
 	// If we are running in single mode, just return thisNode
 	if Mode == ModeSingle {
 		return []Node{thisNode}
 	}
 
-	membersMap := make(map[int32][]Node)
+	// store the available nodes for each partition, grouped by
+	// priority
+	membersMap := make(map[int32]*partitionCandidates)
 	if thisNode.IsReady() {
```

Comment:
how concerned should we be for a scenario like this:
something goes wrong with kafka consumption -> all MT instances start to lag -> all mark themselves as not ready -> cluster is unavailable for reads. whereas it could just have shown some data with the most recent data missing, now users can't see anything

so we basically have a semantical difference:
A) healthy enough for the health check in k8s = in great health = other, older nodes can be downed.
B) healthy enough to participate in data read path = in mediocre health = maybe the data is laggy, but it's better then nothing. if we need to read from this node because it has the best prio, so be it. it's better than showing no data. 

note that for loadbalancers (and just loadbalancing, not deployments), they should also use B for the same reason (better partial data than no data, priority system takes care of the rest - making sure we select the node with best data)

perhaps we should turn this into two different concepts.  is it possible to use a http endpoint for the rolling upgrade health, and a different one for loadbalancing?
what would be really nice in k8s is that if it could compare the priority of instances. then there could be a rule that says if the new node has better priority than the old, the old can be killed.